jsapp/js/components/permissions/userAssetPermsEditor.es6

```diff
@@ -0,0 +1,468 @@
+  onUserExistsStoreChange(result) {
+    // check partial view users
+    const submissionsViewPartialUsers = this.state.submissionsViewPartialUsers;
+    submissionsViewPartialUsers.forEach((username) => {
+      if (result[username] === false) {
+        submissionsViewPartialUsers.pop(submissionsViewPartialUsers.indexOf(username));
+        this.notifyUnknownUser(username);
+      }
+    });
+    this.setState({submissionsViewPartialUsers: submissionsViewPartialUsers});
+
+    // check username
+    if (result[this.state.username] === false) {
+      this.notifyUnknownUser(this.state.username);
+      this.setState({username: ''});
+    }
+
+    // clean usernamesBeingChecked array
+    Object.keys(result).forEach((username) => {
+      const usernamesBeingChecked = this.state.usernamesBeingChecked;
+      usernamesBeingChecked.delete(username);
+      this.setState({usernamesBeingChecked: usernamesBeingChecked});
+    });
+  }
+
+  /**
+   * Blocks submitting non-ready form.
+   */
+  isSubmitEnabled() {
+    const isAnyCheckboxChecked = (
+      this.state.formView ||
+      this.state.formEdit ||
+      this.state.submissionsView ||
+      this.state.submissionsViewPartial ||
+      this.state.submissionsAdd ||
+      this.state.submissionsEdit ||
+      this.state.submissionsValidate
+    );
+    return (
+      isAnyCheckboxChecked &&
+      !this.state.isSubmitPending &&
+      !this.state.isEditingUsername &&
+      this.state.username.length > 0 &&
+      this.state.usernamesBeingChecked.size === 0
+    );
+  }
+
+  getFormData() {
+    return {
+      username: this.state.username,
+      formView: this.state.formView,
+      formEdit: this.state.formEdit,
+      submissionsView: this.state.submissionsView,
+      submissionsViewPartial: this.state.submissionsViewPartial,
+      submissionsViewPartialUsers: this.state.submissionsViewPartialUsers,
+      submissionsAdd: this.state.submissionsAdd,
+      submissionsEdit: this.state.submissionsEdit,
+      submissionsValidate: this.state.submissionsValidate
+    };
+  }
+
+  submit() {
+    if (!this.isSubmitEnabled()) {
+      return;
+    }
+
+    const formData = this.getFormData();
+    const parsedUser = permParser.parseFormData(formData);
+
+    if (parsedUser.length > 0) {
+      // bulk endpoint needs all other users permissions to be passed
+      let otherUserPerms = this.props.nonOwnerPerms.filter((perm) => {
+        return perm.user !== buildUserUrl(formData.username);
+      });
+      this.setState({isSubmitPending: true});
+      actions.permissions.bulkSetAssetPermissions(
+        this.props.uid,
+        otherUserPerms.concat(parsedUser)
+      );
+    } else {
+      // if nothing changes but user submits, just notify parent we're good
+      this.notifyParentAboutSubmitEnd(true);
+    }
+  }
+
+  render() {
+    const isNew = typeof this.props.username === 'undefined';
+
+    const submissionsViewPartialUsersInputProps = {
+      placeholder: t('Add username(s)')
```

Comment:
It's either TAB or ENTER, so maybe "Type a username and press Tab/Enter"? 😆I could change the label if there are some usernames already typed in, like make it empty?

Reply:
hehe, I didn't realize that Enter also worked—and Enter is sometimes called Return. Plus, mobile devices might not have an easily-accessible Tab key. On my Android phone, I have a "Go" button on my keyboard while typing a username, and that does work.

Since usernames aren't allowed to have spaces, could we make space the separator instead of Tab/Enter? Then we could say something like "Enter usernames separated by spaces", which seems pretty universal. We should probably use the on-blur handler, too, if we go that way since adding a space after a single username wouldn't make sense.

PS: "Select fields subset" isn't abundantly clear, and neither is "Add tag(s)", but I guess people haven't complained yet—or they just don't use the features :scream_cat:

Reply:
Changed the label and added spacebar to keys